lib/hackmode/realmode.js

```diff
@@ -0,0 +1,1159 @@
+            if (this._hasMeta(dnaAddress, metaId)) {
+              log.t('metaContent is known:', metaContent)
+              continue
+            }
+            log.t('metaContent is unknown:', metaContent)
+            this._bookkeepAddress(this._storedMetaBook, dnaAddress, metaId)
+            toStoreList.push(metaContent)
+          }
+          // Send toStoreList to Core
+          // log.t(this.me() + ' Sending IPC handleStoreMeta: ', toStoreList)
+          this._ipcSend('json', {
+            method: 'handleStoreMeta',
+            dnaAddress,
+            providerAgentId: data.providerAgentId,
+            entryAddress: data.entryAddress,
+            attribute: data.attribute,
+            contentList: toStoreList
+          })
+        }
+      })
+      this._memory[dnaAddress] = mem
+    }
+    return this._memory[dnaAddress]
+  }
+
+  /**
+   *  Check if agent is tracking dna.
+   *  If not, will try to send a FailureResult back to sender (if sender info is provided).
+   *  Returns transportId of receiverAgentId if agent is tracking dna.
+   *  @private
+   */
+  _getTransportIdOrFail (dnaAddress, receiverAgentId, senderAgentId, requestId) {
+    const peer = this._getDhtRef(dnaAddress).getPeerLocal(receiverAgentId)
+    if (!peer) {
+      // Send FailureResult back to IPC, should be senderAgentId
+      log.e(this.me() + ' #### CHECK FAILED for (agent) "' + receiverAgentId + '" + (DNA) "' + dnaAddress + '" (sender: ' + senderAgentId + ')')
+      this._ipcSend('json', {
+        method: 'failureResult',
+        dnaAddress: dnaAddress,
+        _id: requestId,
+        toAgentId: senderAgentId,
+        errorInfo: 'No routing for agent id "' + receiverAgentId + '"'
+      })
+      return null
+    }
+    const peerId = peer.peerTransport.substring(5) // remove 'hc://'
+    log.t(this.me() + ' oooo CHECK OK for (agent)"' + receiverAgentId + '" + (DNA) "' + dnaAddress + '" = ' + this.nick(peerId))
+    return peerId
+  }
+
+  /**
+   * @private
+   */
+  _untrack (dnaAddress, agentId) {
+    log.t(this.me() + ' _untrack() for "' + agentId + '" for DNA "' + dnaAddress + '"')
+    this._ipcRemoveTrack(agentId, dnaAddress)
+  }
+
+  /**
+   * @private
+   */
+  async _track (dnaAddress, agentId) {
+    log.t(this.me() + ' _track', dnaAddress, agentId)
+    // Bookkeep agentId -> dnaAddress
+    this._ipcAddTrack(agentId, dnaAddress)
+    // Init DHT for this DNA and bookkeep agentId -> PeerId for self
+    if (!(dnaAddress in this._dhtPerDna)) {
+      await this._initDht(dnaAddress, agentId)
+    } else {
+      const peerHoldEvent = DhtEvent.peerHoldRequest(
+        agentId,
+        'hc://' + this._p2p.getId(),
+        Buffer.from('').toString('base64'),
+        Date.now())
+      this._handleDhtEvent(peerHoldEvent, agentId)
+    }
+
+    // Make sure thisPeer data is stored
+    const dht = this._getDhtRef(dnaAddress)
+    let time = 0
+    while (!dht.getPeerLocal(agentId) && time < 1000) {
```

Comment:
d'oh! I wonder if there's something inside the dht code we could tweak to make this work without the polling... Another thing we can look into later.

Reply:
The polling is needed because internally the fulldht adds a peerHoldRequest task of thisPeer during init. The only tweak I see would be to execute that task directly instead of adding it to the task stack.